eternalcombat-api/src/main/java/com/eternalcode/combat/fight/FightManager.java

```diff
@@ -52,7 +52,7 @@ public FightTagEvent tag(UUID target, Duration delay, CauseOfTag causeOfTag) {
         Instant now = Instant.now();
         Instant endOfCombatLog = now.plus(delay);
 
-        FightTag fightTag = new FightTag(target, endOfCombatLog);
+        FightTag fightTag = new FightTag(target, endOfCombatLog, tagger);
```

Comment:
tutaj bym sprawdzał czy tagger jest null (jeśli jest nullable coś to oznaczaj @Nullable bo teraz sam nie do końca wiem jakie były zamiary kodu) jeśli jest null to pobierasz starego taga i szukasz last taggera. Jeśli tagger będzie notnull to nic nie trzeba zmieniać